### PR TITLE
Support collapsible groups of properties

### PR DESCRIPTION
**Pull request by jourdain**
Additionally, keep track of collapse state by property id so that
we always remember which individual groups were opened and closed.

### CODE REVIEW

**Comment by scottwittenburg** [comment]
@jourdain Not sure if you already had a chance to review this?

File: src/panels/ControlPanel/PipelineBrowser/index.js [hunk 1]
```diff
@@ -42,6 +42,7 @@ export const PipelineBrowser = React.createClass({
     updatePreset: React.PropTypes.func,
     updateScalarRange: React.PropTypes.func,
     updateCollapsableState: React.PropTypes.func,
+    updateGroupCollapsibleState: React.PropTypes.func,
```
Comment (jourdain) [inline src/panels/ControlPanel/PipelineBrowser/index.js]: Does not looks like this is used?

File: src/redux/selectors/proxies.js [hunk 2]
```diff
@@ -2,6 +2,31 @@ import { createSelector } from 'reselect';
...
+    }
+  });
+}
+
+function decorateCollapsibleGroups(proxy, groupCollapseState) {
+  checkPropertyList(proxy.properties, proxy.ui, groupCollapseState);
+  return proxy;
+}
+
+// ----------------------------------------------------------------------------
```
Comment (jourdain) [inline src/redux/selectors/proxies.js]: As you can see processing the data that way was not too bad instead of doing in the widget...

**Comment by jourdain** [comment]
LGTM except for the `src/panels/ControlPanel/PipelineBrowser/index.js`

**Pull request merged**